doc/man3/OSSL_CMP_log_open.pod

```diff
@@ -0,0 +1,152 @@
+
+ typedef void (*OSSL_cmp_log_cb_t)(const char *component,
+                                   const char *file, int line,
+                                   OSSL_CMP_severity level, const char *msg);
+
+The parameters may provide
+a component identifier (which may be a library name or function name) or NULL,
+a file path name or NULL,
+a line number or 0 indicating the source code location,
+a severity level, and
+a message string describing the nature of the event.
+
+Even when an activity is successful some warnings may be useful and some degree
+of auditing may be required. Therefore the logging facility supports a severity
+level and the callback function has a B<level> parameter indicating such a
+level, such that error, warning, info, debug, etc. can be treated differently.
+
+The callback function may itself do non-trivial tasks like writing to
+a log file or remote stream, which in turn may fail.
+Therefore the function should return 1 on success and 0 on failure.
+
+OSSL_CMP_log_open() initializes the CMP-specific logging facility to output
+by default to B<stdout>. It fails if the integrated tracing is disabled or STDIO
+is not available. It should be called at least once during application startup.
+It may be called multiple times.
+
+OSSL_CMP_log_close() should be called when all activities are finished to flush
+any pending CMP-specific log output and deallocate related resources.
+It may be called multiple times.
+
+OSSL_CMP_alert() outputs a simple alert message via the trace API.
+OSSL_CMP_err() outputs a simple error message via the trace API.
+OSSL_CMP_warn() outputs a simple warning message via the trace API.
+OSSL_CMP_info() outputs a simple info message via the trace API.
+OSSL_CMP_debug() outputs a simple debug message via the trace API.
+
+Note that due to the design of the trace API used, the log functions have no
+effect unless the B<enable-trace> option is used during build configuration.
+
+OSSL_CMP_add_error_txt() appends text to the extra data field of the last
+error message in the OpenSSL error queue, after adding the optional separator
+unless data has been empty so far. The text can be of arbitrary length,
+which is not possible when using L<ERR_add_error_data(3)> in conjunction with
+L<ERR_print_errors_cb(3)>.
+
+OSSL_CMP_add_error_data() is a macro calling
+B<OSSL_CMP_add_error_txt()> with the separator being ":".
+
+OSSL_CMP_add_error_line() is a macro calling
+B<OSSL_CMP_add_error_txt()> with the separator being "\n".
+
+OSSL_CMP_print_errors_cb() outputs any entries in the OpenSSL error queue
+similarly to B<ERR_print_errors_cb()>, but using the CMP log callback function
+C<log_fn> for uniformity with CMP logging if given. Otherwise it uses
+B<ERR_print_errors(3)> to print to STDERR (unless OPENSSL_NO_STDIO is defined).
+
+=head1 RETURN VALUES
+
+OSSL_CMP_log_close(), OSSL_CMP_add_error_txt(), and OSSL_CMP_print_errors_cb()
+do not return anything.
+
+All other functions return 1 on success, 0 on error.
```

Comment:
We should have a HISTORY section for all the public API man pages introduced here.

Reply:
Fixed